engine/src/lib.rs

```diff
@@ -204,33 +219,41 @@ mod contract {
     ///
     /// [`paused`]: crate::contract::pause_precompiles
     #[no_mangle]
+    #[named]
     pub extern "C" fn resume_precompiles() {
-        let io = Runtime;
+        let mut io = Runtime;
         let state = state::get_state(&io).sdk_unwrap();
         let predecessor_account_id = io.predecessor_account_id();
 
         require_owner_only(&state, &predecessor_account_id);
 
-        let args: PausePrecompilesCallArgs = io.read_input_borsh().sdk_unwrap();
+        let input = io.read_input();
+        let args: PausePrecompilesCallArgs = input.to_value().sdk_unwrap();
         let flags = PrecompileFlags::from_bits_truncate(args.paused_mask);
         let mut pauser = EnginePrecompilesPauser::from_io(io);
         pauser.resume_precompiles(flags);
+
+        update_hashchain(&mut io, function_name!(), &input.to_vec(), &[0; 0]);
```

Comment:
In many methods, I'm leaving the output empty (`&[0; 0]`) because I don't see a call to `sdk_process`. Please check this list of methods and let me know if I'm missing the output for some:

1. resume_precompiles
2. pause_precompiles
3. register_relayer
4. factory_update
5. factory_update_address_version
6. factory_set_wnear_address
7. ft_on_transfer
8. refund_on_error
9. new_eth_connector
10. set_eth_connector_contract_data
11. deposit
12. finish_deposit
13. ft_transfer
14. ft_resolve_transfer
15. ft_transfer_call
16. storage_deposit
17. storage_unregister
18. storage_withdraw
19. get_paused_flags

Reply:
Methods that include `promise_return` are tricky. The "output" from this function on the Near level is a receipt ID because the function is making a cross-contract call. Unfortunately for us the receipt ID cannot be known inside our contract so we cannot capture it for the hashchain. This means it will still be possible for someone to lie about what receipts corresponded with some executions, but I do not see a good way to resolve this problem.

This applies to functions such as `deposit`, `finish_deposit` and `ft_transfer_call`.

For functions where there is legitimately no return value I think `&[]` is fine. The `&[0; 0]` I think is equivalent, but harder to read in my opinion.

> 19. get_paused_flags

I think you means `set_paused_flags`.

Reply:
Regarding `promise_return`, what would happen then on the Refiner for the corresponding txs? We would have there the receipt ID I suppose, so we will need to ignore it to build the same hash as in the Engine.

Sure, `&[] `is better.

And yes, I mean `set_paused_flags`.

Reply:
Yes that's right, the Refiner would have access to the receipt ID, but would have to ignore it to get the same result as on-chain.